### PR TITLE
podvm: bump RHEL version to 9.4

### PR DESCRIPTION
**Pull request by snir911**

### CODE REVIEW

**Review by bpradipt — APPROVED** [review]
/lgtm

**Comment by stevenhorsman** [comment]
@Saripalli-lavanya - do you know if there are are issues with RHEL 9.4 on s390x? Thanks

**Comment by bpradipt** [comment]
> @Saripalli-lavanya - do you know if there are are issues with RHEL 9.4 on s390x? Thanks

The current changes are specific for x86_64. It would be good to have it consistent with s390x. Usually minor release version bump for distro shouldn't create any incompatibility issues.

**Comment by Saripalli-lavanya** [comment]
> @Saripalli-lavanya - do you know if there are are issues with RHEL 9.4 on s390x? Thanks

Hi @stevenhorsman, I'm encountering a similar error to what I saw with 'UBI:9.3'.

```
STEP 13/26: RUN if [[ -n "${ACTIVATION_KEY}" && -n "${ORG_ID}" ]]; then     subscription-manager register --org=${ORG_ID} --activationkey=${ACTIVATION_KEY};     fi
subscription-manager is operating in container mode. Use your host system to manage subscriptions.

Error: building at STEP "RUN if [[ -n "${ACTIVATION_KEY}" && -n "${ORG_ID}" ]]; then     subscription-manager register --org=${ORG_ID} --activationkey=${ACTIVATION_KEY};     fi": while running runtime: exit status 78
make: *** [Makefile:168: podvm-builder] Error 78
[root@a3elp36 cloud-api-adaptor]# cat ./podvm/Dockerfile.podvm_builder.rhel
```

**Comment by bpradipt** [comment]
> > @Saripalli-lavanya - do you know if there are are issues with RHEL 9.4 on s390x? Thanks
> 
> Hi @stevenhorsman, I'm encountering a similar error to what I saw with 'UBI:9.3'.
> 
> ```
> STEP 13/26: RUN if [[ -n "${ACTIVATION_KEY}" && -n "${ORG_ID}" ]]; then     subscription-manager register --org=${ORG_ID} --activationkey=${ACTIVATION_KEY};     fi
> subscription-manager is operating in container mode. Use your host system to manage subscriptions.
> 
> Error: building at STEP "RUN if [[ -n "${ACTIVATION_KEY}" && -n "${ORG_ID}" ]]; then     subscription-manager register --org=${ORG_ID} --activationkey=${ACTIVATION_KEY};     fi": while running runtime: exit status 78
> make: *** [Makefile:168: podvm-builder] Error 78
> [root@a3elp36 cloud-api-adaptor]# cat ./podvm/Dockerfile.podvm_builder.rhel
> ```

@Saripalli-lavanya there are two aspects - one is the base image for dockerfile  and another is the RHEL 9.x image used for podVM.
Can you run a standalone ubi:9.4 or ubi:9.3 container and run the subscription manager and see what's the error.

**Comment by Saripalli-lavanya** [comment]
> > > @Saripalli-lavanya - do you know if there are are issues with RHEL 9.4 on s390x? Thanks
> > 
> > 
> > Hi @stevenhorsman, I'm encountering a similar error to what I saw with 'UBI:9.3'.
> > ```
> > STEP 13/26: RUN if [[ -n "${ACTIVATION_KEY}" && -n "${ORG_ID}" ]]; then     subscription-manager register --org=${ORG_ID} --activationkey=${ACTIVATION_KEY};     fi
> > subscription-manager is operating in container mode. Use your host system to manage subscriptions.
> > 
> > Error: building at STEP "RUN if [[ -n "${ACTIVATION_KEY}" && -n "${ORG_ID}" ]]; then     subscription-manager register --org=${ORG_ID} --activationkey=${ACTIVATION_KEY};     fi": while running runtime: exit status 78
> > make: *** [Makefile:168: podvm-builder] Error 78
> > [root@a3elp36 cloud-api-adaptor]# cat ./podvm/Dockerfile.podvm_builder.rhel
> > ```
> 
> @Saripalli-lavanya there are two aspects - one is the base image for dockerfile and another is the RHEL 9.x image used for podVM. Can you run a standalone ubi:9.4 or ubi:9.3 container and run the subscription manager and see what's the error.
> 
> ```
> docker run -it registry.access.redhat.com/ubi9/ubi:9.4 bash
> 
> subscription-manager register --org=<orgid> --activationkey=<activationkey>
> ```

@sure @bpradipt, I see below outputs, for ubi:9.4

```
[root@a3elp36 cloud-api-adaptor]# docker run -it registry.access.redhat.com/ubi9/ubi:9.4 bash 
Emulate Docker CLI using podman. Create /etc/containers/nodocker to quiet msg.
[root@e7690caa769c /]# subscription-manager register --org=1**** --activationkey="RHEL***"
subscription-manager is operating in container mode. Use your host system to manage subscriptions.
[root@e7690caa769c /]# uname -a
Linux e7690caa769c 5.14.0-362.8.1.el9_3.s390x #1 SMP Tue Oct 3 09:00:29 EDT 2023 s390x s390x s390x GNU/Linux
```

for 9.3

```
[root@a3elp36 cloud-api-adaptor]# docker run -it registry.access.redhat.com/ubi9/ubi:9.3 bash 
Emulate Docker CLI using podman. Create /etc/containers/nodocker to quiet msg.
Trying to pull registry.access.redhat.com/ubi9/ubi:9.3...
Getting image source signatures
Checking if image destination supports signatures
Copying blob 2a24740263c4 done   | 
Copying config fec52210cf done   | 
Writing manifest to image destination
Storing signatures
[root@9e2013375ee2 /]# subscription-manager register --org=1979710 --activationkey="RHEL-TEST"
subscription-manager is operating in container mode. Use your host system to manage subscriptions.

[root@9e2013375ee2 /]# uname -a
Linux 9e2013375ee2 5.14.0-362.8.1.el9_3.s390x #1 SMP Tue Oct 3 09:00:29 EDT 2023 s390x s390x s390x GNU/Linux

```

for 9.2

```
[root@a3elp36 cloud-api-adaptor]# docker run -it registry.access.redhat.com/ubi9/ubi:9.2 bash 
Emulate Docker CLI using podman. Create /etc/containers/nodocker to quiet msg.
[root@cd091b1f5702 /]# subscription-manager register --org=1979710 --activationkey="RHEL-TEST"
subscription-manager is disabled when running inside a container. Please refer to your host system for subscription management.

[root@cd091b1f5702 /]# uname -a
Linux cd091b1f5702 5.14.0-362.8.1.el9_3.s390x #1 SMP Tue Oct 3 09:00:29 EDT 2023 s390x s390x s390x GNU/Linux
```

**Comment by bpradipt** [comment]
@Saripalli-lavanya so you get exactly the same behaviour with subscription-manager irrespective of the ubi version.

I'm unable to understand how this Dockerfile works (without any changes to ubi version) in s390x.
Have you checked if the original Dockerfile works (without this PR) ?

**Comment by Saripalli-lavanya** [comment]
I have updated the UBI version from 9.4 to 9.2. Below is a snippet of the output

```
[root@a3elp36 cloud-api-adaptor]# PODVM_DISTRO=rhel make podvm-builder podvm-binaries podvm-image ARCH=s390x CLOUD_PROVIDER=libvirt ORG_ID=$ORG_ID ACTIVATION_KEY=$ACTIVATION_KEY PROTOC_VERSION=26.1
docker buildx build -t quay.io/confidential-containers/podvm-builder-rhel:6ea0e23252589bc69bc8a3a804811a258267fd0612865c5f22e7d6f8d81ca151 -f podvm/Dockerfile.podvm_builder.rhel \
--build-arg GO_VERSION=1.21.12 \
--build-arg ORG_ID=1**** \
--build-arg ARCH=s390x \
--build-arg ACTIVATION_KEY=RHEL**** \
--build-arg PACKER_VERSION=v1.9.4 \
--build-arg PROTOC_VERSION=26.1 \
--build-arg RUST_VERSION=1.74.0 \
--build-arg YQ_VERSION=v4.35.1 \
--build-arg YQ_CHECKSUM="sha256:4e6324d08630e7df733894a11830412a43703682d65a76f1fc925aac08268a45" \
--build-arg YQ_ARCH=s390x \
 	--load  . --no-cache
Emulate Docker CLI using podman. Create /etc/containers/nodocker to quiet msg.
STEP 1/26: FROM registry.access.redhat.com/ubi9/ubi:9.2
STEP 2/26: ARG ARCH
--> 089145cf6198
STEP 3/26: ARG PACKER_VERSION
--> 94f2d54d945a
STEP 4/26: ARG GO_VERSION
--> ef235505d942
STEP 5/26: ARG PROTOC_VERSION
--> fc2cd02e176b
STEP 6/26: ARG RUST_VERSION
--> 62665b0b7c5f
STEP 7/26: ARG ORG_ID
--> e6961ddeb80e
STEP 8/26: ARG ACTIVATION_KEY
--> 9c74156afcd0
STEP 9/26: ARG YQ_VERSION
--> 1512b0db346a
STEP 10/26: ARG YQ_CHECKSUM
--> d0bd42b92156
STEP 11/26: ENV SMDEV_CONTAINER_OFF=1
--> 1c1605bffe5d
STEP 12/26: ENV ARCH=${ARCH}
--> 76a3547d26a7
STEP 13/26: RUN if [[ -n "${ACTIVATION_KEY}" && -n "${ORG_ID}" ]]; then     subscription-manager register --org=${ORG_ID} --activationkey=${ACTIVATION_KEY};     fi
The system has been registered with ID: 41efa1c8-5e19-4d78-b938-25127a8738cf
The registered system name is: 7f0db795dcab
--> 0143acabb7a3
STEP 14/26: RUN subscription-manager repos --enable codeready-builder-for-rhel-9-${ARCH}-rpms;     dnf groupinstall -y 'Development Tools';     dnf install -y yum-utils gnupg git --allowerasing curl pkg-config clang perl libseccomp-devel gpgme-devel     device-mapper-devel qemu-kvm unzip wget libassuan-devel genisoimage cloud-utils-growpart cloud-init;
```
Changes made:
```
[root@a3elp36` cloud-api-adaptor]# git status
On branch bump_rhel94
Your branch is up to date with 'origin/bump_rhel94'.

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
	modified:   Makefile
	modified:   podvm/Dockerfile.podvm_builder.rhel

Untracked files:
  (use "git add <file>..." to include in what will be committed)
	podvm/rhel-guest-image-9.2-20230414.17.s390x.qcow2

no changes added to commit (use "git add" and/or "git commit -a")
[root@a3elp36 cloud-api-adaptor]# git diff
diff --git a/src/cloud-api-adaptor/Makefile b/src/cloud-api-adaptor/Makefile
index 73a5004..748695f 100644
--- a/src/cloud-api-adaptor/Makefile
+++ b/src/cloud-api-adaptor/Makefile
@@ -176,7 +176,7 @@ ifeq ($(PODVM_DISTRO),rhel)
        --build-arg YQ_VERSION=$(YQ_VERSION) \
        --build-arg YQ_CHECKSUM=${YQ_CHECKSUM_$(ARCH)} \
        --build-arg YQ_ARCH=$(ARCH) \
-       $(DOCKER_OPTS) .
+       $(DOCKER_OPTS) . --no-cache
 else
        docker buildx build -t $(PODVM_BUILDER_IMAGE) -f podvm/$(BUILDER_DOCKERFILE) \
        --build-arg GO_VERSION=$(GO_VERSION) \
diff --git a/src/cloud-api-adaptor/podvm/Dockerfile.podvm_builder.rhel b/src/cloud-api-adaptor/podvm/Dockerfile.podvm_builder.rhel
index 7de8c64..fa60e89 100644
--- a/src/cloud-api-adaptor/podvm/Dockerfile.podvm_builder.rhel
+++ b/src/cloud-api-adaptor/podvm/Dockerfile.podvm_builder.rhel
@@ -6,7 +6,7 @@
 # Creates a builder container image that should be used to build the Pod VM
 # disk inside a container.
 #
-FROM registry.access.redhat.com/ubi9/ubi:9.4
+FROM registry.access.redhat.com/ubi9/ubi:9.2
 
 ARG ARCH
 ARG PACKER_VERSION
[root@a3elp36 cloud-api-adaptor]# 
```

**Comment by Saripalli-lavanya** [comment]
@bpradipt, in addition to the above comment.

with UBI:9.4 as base image, I've commented below lines from `Dockerfile.podvm_builder.rhel` file post which i was able to proceed further. but eventually failed with another error.

```
+#RUN if [[ -n "${ACTIVATION_KEY}" && -n "${ORG_ID}" ]]; then \
+ #   subscription-manager register --org=${ORG_ID} --activationkey=${ACTIVATION_KEY}; \
+  #  fi
```

```
--> f8d805cf8d6d
STEP 25/25: ENV=GOPATH /src
ERRO[0281] +(UNHANDLED LOGLEVEL) &imagebuilder.Step{Env:[]string{"PROTOC_VERSION=26.1", "ARCH=s390x", "PACKER_VERSION=v1.9.4", "ORG_ID=1****", "YQ_VERSION=v4.35.1", "YQ_CHECKSUM=sha256:4e6324d08630e7df733894a11830412a43703682d65a76f1fc925aac08268a45", "GO_VERSION=1.21.12", "RUST_VERSION=1.74.0", "ACTIVATION_KEY=RHEL****", "PATH=/root/.cargo/bin:/usr/local/go/bin:/usr/local/go/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin", "container=oci", "SMDEV_CONTAINER_OFF=1"}, Command:"env=gopath", Args:[]string{""}, Flags:[]string{}, Attrs:map[string]bool(nil), Message:"ENV=GOPATH ", Heredocs:[]parser.Heredoc(nil), Original:"ENV=GOPATH /src"} 
Error: building at STEP "ENV=GOPATH ": Build error: Unknown instruction: "ENV=GOPATH" &imagebuilder.Step{Env:[]string{"PROTOC_VERSION=26.1", "ARCH=s390x", "PACKER_VERSION=v1.9.4", "ORG_ID=1*****", "YQ_VERSION=v4.35.1", "YQ_CHECKSUM=sha256:4e6324d08630e7df733894a11830412a43703682d65a76f1fc925aac08268a45", "GO_VERSION=1.21.12", "RUST_VERSION=1.74.0", "ACTIVATION_KEY=RHEL****", "PATH=/root/.cargo/bin:/usr/local/go/bin:/usr/local/go/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin", "container=oci", "SMDEV_CONTAINER_OFF=1"}, Command:"env=gopath", Args:[]string{""}, Flags:[]string{}, Attrs:map[string]bool(nil), Message:"ENV=GOPATH ", Heredocs:[]parser.Heredoc(nil), Original:"ENV=GOPATH /src"}
make: *** [Makefile:168: podvm-builder] Error 125

```
with UBI:9.2 if i comment same subscription-manager register lines i see below failure.
```
STEP 13/25: RUN subscription-manager repos --enable codeready-builder-for-rhel-9-${ARCH}-rpms;     dnf groupinstall -y 'Development Tools';     dnf install -y yum-utils gnupg git --allowerasing curl pkg-config clang perl libseccomp-devel gpgme-devel     device-mapper-devel qemu-kvm unzip wget libassuan-devel genisoimage cloud-utils-growpart cloud-init; 
This system has no repositories available through subscriptions.
Updating Subscription Management repositories.
Unable to read consumer identity

This system is not registered with an entitlement server. You can use subscription-manager to register.

Red Hat Universal Base Image 9 (RPMs) - BaseOS  848 kB/s | 419 kB     00:00    
Red Hat Universal Base Image 9 (RPMs) - AppStre 5.4 MB/s | 1.8 MB     00:00    
Red Hat Universal Base Image 9 (RPMs) - CodeRea 603 kB/s | 166 kB     00:00    
Module or Group 'Development Tools' is not available.
Error: Nothing to do.
Updating Subscription Management repositories.
Unable to read consumer identity

This system is not registered with an entitlement server. You can use subscription-manager to register.

Package gnupg2-2.3.3-2.el9_0.s390x is already installed.
No match for argument: qemu-kvm
No match for argument: genisoimage
No match for argument: cloud-utils-growpart
No match for argument: cloud-init
Error: Unable to find a match: qemu-kvm genisoimage cloud-utils-growpart cloud-init
Error: building at STEP "RUN subscription-manager repos --enable codeready-builder-for-rhel-9-${ARCH}-rpms;     dnf groupinstall -y 'Development Tools';     dnf install -y yum-utils gnupg git --allowerasing curl pkg-config clang perl libseccomp-devel gpgme-devel     device-mapper-devel qemu-kvm unzip wget libassuan-devel genisoimage cloud-utils-growpart cloud-init;": while running runtime: exit status 1
make: *** [Makefile:168: podvm-builder] Error 1
```

Thank you

**Comment by bpradipt** [comment]
@Saripalli-lavanya can you perform these steps in your environment
1. Remove all docker images (`systemctl stop docker && rm -rf /var/lib/docker && systemctl start docker` should do)
2. Register the host system (`subscription-manager register --org=<orgid> --activationkey=<activationkey>`)
3. Unset ORG_ID and ACTIVATION_KEY  (so that the next step doesn't do subscription-register)
4. Run docker build  (use ubi:9.4) as the base

**Comment by bpradipt** [comment]
@snir911 I realised there is another existing typo in Dockerfile.podvm_builder.rhel
```
ENV=GOPATH /src
```

It should be 
```
ENV GOPATH=/src
```

I wonder how the existing Dockerfile.podvm_builder.rhel even worked :-)

**Comment by snir911** [comment]
> @snir911 I realised there is another existing typo in Dockerfile.podvm_builder.rhel

Nice catch, thanks, I've pushed again with the typo fix

> docker run -it registry.access.redhat.com/ubi9/ubi:9.4 bash
subscription-manager register --org=<orgid> --activationkey=<activationkey>

The "right" way to run this would be to run on a registered host system, however, setting the `SMDEV_CONTAINER_OFF=1` environment variable should allow you to register manually on any system (as we did in the Dockerfile )

Having said that, it seems like it isn't supported anymore in latest [subscription-manager](https://github.com/candlepin/subscription-manager/commit/5aacb8b3ed0287c4ee68b817f40da5700dfb5c5d) version, I pushed another commit that should fix it, @Saripalli-lavanya could you please check if it helps.

**Comment by snir911** [comment]
I pushed couple of changes, however, it seems to me we have wrongly implemented this, IIUC environment variable values provided through make ([and passed to docker using --build-arg](https://github.com/confidential-containers/cloud-api-adaptor/blob/main/src/cloud-api-adaptor/Makefile#L182-L187)) are being ignored at build time [because of the default values we set in the Dockerfiles](https://github.com/confidential-containers/cloud-api-adaptor/blob/55905cdbafb70e1a6a8032fff1d73768f0993af1/src/cloud-api-adaptor/podvm/Dockerfile.podvm_builder#L11-L21) ; i.e. all builder image builds done using make are used with the default values (x86 etc..), can someone else confirm? or am i missing something?

Would it make sense to simply drop defaults from Dockerfiles?

**Comment by Saripalli-lavanya** [comment]
Great @snir911!  yes, it is working now , thank you.

```
[root@a3elp36 cloud-api-adaptor]# podman images
REPOSITORY                                                           TAG                                                               IMAGE ID      CREATED             SIZE
quay.io/confidential-containers/podvm-libvirt-rhel-s390x             6ea0e23252589bc69bc8a3a804811a258267fd0612865c5f22e7d6f8d81ca151  32d813157cb5  40 seconds ago      1.03 GB
<none>                                                               <none>                                                            4c33cfdbeb60  About a minute ago  10.9 GB
quay.io/confidential-containers/podvm-binaries-rhel-s390x            6ea0e23252589bc69bc8a3a804811a258267fd0612865c5f22e7d6f8d81ca151  56a5a2136947  8 minutes ago       299 MB
<none>                                                               <none>                                                            ad2d74b12af3  9 minutes ago       14.8 GB
quay.io/confidential-containers/podvm-builder-rhel                   6ea0e23252589bc69bc8a3a804811a258267fd0612865c5f22e7d6f8d81ca151  327b0c61d66f  28 minutes ago      7.19 GB

```

**Comment by snir911** [comment]
@Saripalli-lavanya can you check if the built binaries are actually s390x binaries i suspect all the values you set are ignored so it's actually x86

**Comment by bpradipt** [comment]
> > @snir911 I realised there is another existing typo in Dockerfile.podvm_builder.rhel
> 
> Nice catch, thanks, I've pushed again with the typo fix
> 
> > docker run -it registry.access.redhat.com/ubi9/ubi:9.4 bash
> > subscription-manager register --org= --activationkey=
> 
> The "right" way to run this would be to run on a registered host system, however, setting the `SMDEV_CONTAINER_OFF=1` environment variable should allow you to register manually on any system (as we did in the Dockerfile )
> 
> Having said that, it seems like it isn't supported anymore in latest [subscription-manager](https://github.com/candlepin/subscription-manager/commit/5aacb8b3ed0287c4ee68b817f40da5700dfb5c5d) version, I pushed another commit that should fix it, @Saripalli-lavanya could you please check if it helps.

I could run the subscription-manager command successfully inside a container on a Fedora host (non-registered system) and it worked without any issues. This was without using the changes in the current PR

**Comment by bpradipt** [comment]
> I pushed couple of changes, however, it seems to me we have wrongly implemented this, IIUC environment variable values provided through make ([and passed to docker using --build-arg](https://github.com/confidential-containers/cloud-api-adaptor/blob/main/src/cloud-api-adaptor/Makefile#L182-L187)) are being ignored at build time [because of the default values we set in the Dockerfiles](https://github.com/confidential-containers/cloud-api-adaptor/blob/55905cdbafb70e1a6a8032fff1d73768f0993af1/src/cloud-api-adaptor/podvm/Dockerfile.podvm_builder#L11-L21) ; i.e. all builder image builds done using make are used with the default values (x86 etc..), can someone else confirm? or am i missing something?
> 
> Would it make sense to simply drop defaults from Dockerfiles?

Defaults will be used when it's not set in build-args. https://docs.docker.com/reference/dockerfile/#default-values
Probably something else wrong in your environment?
You can also do a small test
```
FROM ubuntu:22.04

ARG test=hello

RUN echo ${test}
```

Sample runs

```

# docker build -t test -f Dockerfile .
[+] Building 0.6s (7/7) FINISHED                                                                                                                                              docker:default
 => [internal] load build definition from Dockerfile                                                                                                                                    0.0s
 => => transferring dockerfile: 149B                                                                                                                                                    0.0s
 => [internal] load metadata for docker.io/library/ubuntu:22.04                                                                                                                         0.4s
 => [auth] library/ubuntu:pull token for registry-1.docker.io                                                                                                                           0.0s
 => [internal] load .dockerignore                                                                                                                                                       0.0s
 => => transferring context: 2B                                                                                                                                                         0.0s
 => CACHED [1/2] FROM docker.io/library/ubuntu:22.04@sha256:340d9b015b194dc6e2a13938944e0d016e57b9679963fdeb9ce021daac430221                                                            0.0s
 => [2/2] RUN echo hello                                                                                                                                                                0.2s
 => exporting to image                                                                                                                                                                  0.0s
 => => exporting layers                                                                                                                                                                 0.0s
 => => writing image sha256:19c7c6bbaec76a99f7b8df1467a879ed6632378085f2e1cb24799a102c2b1bae                                                                                            0.0s
 => => naming to docker.io/library/test                                                                                                                                                 0.0s


# docker build -t test --build-arg test=bye -f Dockerfile .
[+] Building 0.3s (6/6) FINISHED                                                                                                                                              docker:default
 => [internal] load build definition from Dockerfile                                                                                                                                    0.0s
 => => transferring dockerfile: 149B                                                                                                                                                    0.0s
 => [internal] load metadata for docker.io/library/ubuntu:22.04                                                                                                                         0.1s
 => [internal] load .dockerignore                                                                                                                                                       0.0s
 => => transferring context: 2B                                                                                                                                                         0.0s
 => CACHED [1/2] FROM docker.io/library/ubuntu:22.04@sha256:340d9b015b194dc6e2a13938944e0d016e57b9679963fdeb9ce021daac430221                                                            0.0s
 => [2/2] RUN echo bye                                                                                                                                                                  0.2s
 => exporting to image                                                                                                                                                                  0.0s
 => => exporting layers                                                                                                                                                                 0.0s
 => => writing image sha256:d507351e7bf390121b0faaaefaefb26cc87f047662050aae93e87f5dc99a92db                                                                                            0.0s
 => => naming to docker.io/library/test

File: src/cloud-api-adaptor/versions.yaml [hunk 1]
```diff
@@ -12,8 +12,8 @@ cloudimg:
   rhel:
     9: # dummy link, get trial image from: https://www.redhat.com/en/technologies/linux-platforms/enterprise-linux/server/trial
       amd64:
-        url: https://access.redhat.com/downloads/content/479/ver=/rhel---9/9.2/x86_64/product-software
-        checksum: "sha256:34ff41b5274692c984e3860b21136af8b6ae502744c6c7578dda82002fba0287"
+        url: https://access.redhat.com/downloads/content/479/ver=/rhel---9/9.4/x86_64/product-software
+        checksum: "sha256:d362e72518a2d7415d850b8177c814f6fd87f42ca1640bda17e98855e1d6baad"
       s390x:
         url: https://access.redhat.com/downloads/content/433/ver=/rhel---9/9.3%20Beta/s390x/product-software
```
Comment (bpradipt) [inline src/cloud-api-adaptor/versions.yaml:18]: Should the s390x RHEL image be switche to 9.4 also?
@stevenhorsman @Saripalli-lavanya ??

**Comment by Saripalli-lavanya** [comment]
> @Saripalli-lavanya can you check if the built binaries are actually s390x binaries i suspect all the values you set are ignored so it's actually x86

I've checked for couple of them and I see binaries built are s390x only, please let me know any additional observations or considerations.
```
[root@a3elp36 cloud-api-adaptor]# podman run -it quay.io/confidential-containers/podvm-binaries-rhel-s390x:6ea0e23252589bc69bc8a3a804811a258267fd0612865c5f22e7d6f8d81ca151 /bin/bash
[root@c09eb077ca14 /]# yum install file
[root@c09eb077ca14 /]# tar xzvf /podvm-binaries.tar.gz
[root@c09eb077ca14 /]# cat etc/systemd/system/multi-user.target.wants/agent-protocol-forwarder.service
[Unit]
Description=Agent Protocol Forwarder
After=kata-agent.service
Wants=kata-agent.service
DefaultDependencies=no


[Service]
Type=notify
EnvironmentFile=-/etc/default/agent-protocol-forwarder
ExecStart=/usr/local/bin/agent-protocol-forwarder -kata-agent-namespace /run/netns/podns -kata-agent-socket /run/kata-containers/agent.sock $TLS_OPTIONS $OPTIONS
Restart=on-failure
RestartSec=5s

[Install]
WantedBy=multi-user.target
[root@c09eb077ca14 /]# file /usr/local/bin/agent-protocol-forwarder
/usr/local/bin/agent-protocol-forwarder: ELF 64-bit MSB executable, IBM S/390, version 1 (SYSV), statically linked, Go BuildID=n2_M4dmhURZJAHgpLG9h/6nRI2pmqxD8V3qeQut7L/5Mo8121s85PDwGoG9AQq/Iws7VOAcy7u-_F8avuaf, with debug_info, not stripped
[root@c09eb077ca14 /]# uname
Linux
[root@c09eb077ca14 /]# uname -m
s390x
[root@c09eb077ca14 /]# [root@c09eb077ca14 /]# /usr/local/bin/agent-protocol-forwarder -version
agent-protocol-forwarder version unknown
  commit: c37818796f88fd2b929358644ea09beae4f6b6d7-dirty
  go: go1.21.12
[root@c09eb077ca14 /]# cat etc/systemd/system/multi-user.target.wants/attestation-agent.service
[Unit]
Description=Attestation Agent TTRPC API Server
After=network.target cloud-init.service process-user-data.service

[Service]
Type=simple
ExecStartPre=mkdir -p /run/confidential-containers/attestation-agent
ExecStart=/bin/bash -c \
    'if [ -f /run/peerpod/aa.toml ]; \
    then /usr/local/bin/attestation-agent -c /run/peerpod/aa.toml; \
    else /usr/local/bin/attestation-agent; \
    fi'
RestartSec=1
Restart=always

[Install]
WantedBy=multi-user.target
[root@c09eb077ca14 /]# file /usr/local/bin/attestation-agent
/usr/local/bin/attestation-agent: ELF 64-bit MSB pie executable, IBM S/390, version 1 (SYSV), dynamically linked, interpreter /lib/ld64.so.1, BuildID[sha1]=a972bb8389252eeae4b96dffb88038858db1e78e, for GNU/Linux 3.2.0, with debug_info, not stripped
[root@c09eb077ca14 /]# /usr/local/bin/attestation-agent --version
attestation-agent 

Commit Hash: df60725afe0ba452a25a740cf460c2855442c49a 
Supported Attesters: 
Token plugins: KBS, COCO_AS
```

Thank you

File: src/cloud-api-adaptor/versions.yaml (hunk 1 above)
Comment (snir911) [reply src/cloud-api-adaptor/versions.yaml:18]: right!, i guess it should

Comment (Saripalli-lavanya) [reply src/cloud-api-adaptor/versions.yaml:18]: Adding a point, currently, we are manually downloading s390x image due to subscription constraints. May be to maintain consistency we can switch.

Comment (snir911) [reply src/cloud-api-adaptor/versions.yaml]: @Saripalli-lavanya all RHEL images are usually downloaded manually, I updated the s390x image checksum, can you confirm it's correct.

Comment (Saripalli-lavanya) [reply src/cloud-api-adaptor/versions.yaml]: Yes 👍

**Review by Saripalli-lavanya — APPROVED** [review]

**Review by stevenhorsman — APPROVED** [review]
Approved based on @Saripalli-lavanya's tests. Thanks!

**Pull request merged**